Simplified/NYPLAccount.m

```diff
@@ -222,7 +222,8 @@ - (void)setLicensor:(NSDictionary *)licensor
 - (void)setAuthorizationIdentifier:(NSString *)identifier
 {
   if(!(identifier)) {
-    @throw NSInvalidArgumentException;
+    NYPLLOG(@"Authorization ID (Barcode String) was nil.");
```

Comment:
What was the motivation for not throwing an exception here? I think the app threw `NSInvalidArgumentException` because calling `setAuthorizationIdentifier:` with `nil` indicated an error elsewhere in the program. Simply doing nothing here would just mask the problem and cause it to show up later on and in a harder to detect way, would it not? My preference would be for the app to crash immediately so we'd be able to know the problem is happening and we'd have a clear stack trace.

Reply:
That's fair. I was moving the nil check perhaps a little too deep. With the current and only usecase for the property, the UX is designed with a nil-case, but someone else may have another need for it  later on. I personally don't trust everyone's OPDS feed to always have this be non nil, so I'll just check before setting it from the feed that it originates from.